deliver/lib/deliver/upload_metadata.rb

```diff
@@ -184,7 +184,7 @@ def load_from_filesystem(options)
           next unless File.exist?(path)
 
           UI.message("Loading '#{path}'...")
-          options[key] ||= {}
+          options[key] = {}
```

Comment:
I agree that this PR still needs some work. 

In my case I don't have a Deliverfile. I'm calling `deliver()` from Fastfile and passing in a `name:` argument, which is set based on environment variables set during `--env` processing. I need this functionality because my app has a different name for alpha, beta, and release builds.

For now I'll just delete metadata/en-US/name.txt so that it doesn't conflict.

But there is still a fundamental problem when calling `deliver() `and passing in the `name:` argument. Fastlane shouldn't fail. I see a few possible behaviors for this condition:

1. The passed in `name:` overrides all languages, in which case a warning should be printed.

2. The passed in `name:` is not used at all, in which case a warning should be printed.

3. Fastlane fails with an error stating that a `name:` argument cannot be specified while simultaneously including a locale specific name.txt.

4. `name:` argument overrides default locale only, all others use locale-specific name.txt.